File: src/main/java/net/flyingfishflash/loremlist/domain/lrmlistitem/LrmListItemRepository.java
package net.flyingfishflash.loremlist.domain.lrmlistitem;

import java.util.List;
import java.util.Set;
import java.util.UUID;
import kotlin.Pair;
import net.flyingfishflash.loremlist.persistence.SuccinctLrmComponentPair;

public interface LrmListItemRepository {
  long countByOwnerAndListId(UUID listId, String listOwner);

  long countByOwnerAndItemId(UUID itemId, String itemOwner);

  void create(UUID listId, UUID itemId);

  List<SuccinctLrmComponentPair> create(Set<Pair<UUID, UUID>> associationCollection);

  LrmListItem findByOwnerAndItemIdAndListIdOrNull(UUID itemId, UUID listId, String owner);

  int removeByOwnerAndItemId(UUID itemId, String owner);

  int removeByOwnerAndListId(UUID listId, String owner);

  int removeByOwnerAndListIdAndItemId(UUID listId, UUID itemId, String owner);

  int updateListId(LrmListItem lrmListItem, UUID destinationListId);

  int updateName(LrmListItem lrmListItem);

  int updateDescription(LrmListItem lrmListItem);

  int updateQuantity(LrmListItem lrmListItem);

  int updateIsItemSuppressed(LrmListItem lrmListItem);
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmlistitem/LrmListItemRepository.kt
package net.flyingfishflash.loremlist.domain.lrmlistitem

import net.flyingfishflash.loremlist.persistence.SuccinctLrmComponentPair
import java.util.*

interface LrmListItemRepository {
  fun countByOwnerAndListId(listId: UUID, listOwner: String): Long
  fun countByOwnerAndItemId(itemId: UUID, itemOwner: String): Long
  fun create(listId: UUID, itemId: UUID)
  fun create(associationCollection: Set<Pair<UUID, UUID>>): List<SuccinctLrmComponentPair>
  fun findByOwnerAndItemIdAndListIdOrNull(itemId: UUID, listId: UUID, owner: String): LrmListItem?
  fun removeByOwnerAndItemId(itemId: UUID, owner: String): Int
  fun removeByOwnerAndListId(listId: UUID, owner: String): Int
  fun removeByOwnerAndListIdAndItemId(listId: UUID, itemId: UUID, owner: String): Int
  fun updateListId(lrmListItem: LrmListItem, destinationListId: UUID): Int
  fun updateName(lrmListItem: LrmListItem): Int
  fun updateDescription(lrmListItem: LrmListItem): Int
  fun updateQuantity(lrmListItem: LrmListItem): Int
  fun updateIsItemSuppressed(lrmListItem: LrmListItem): Int
}
